src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolItem.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    using SymbolKind = LanguageServer.Protocol.SymbolKind;
+
+    internal class DocumentSymbolItem : INotifyPropertyChanged
+    {
+        private bool _isSelected;
+        private bool _isExpanded;
+
+        public string Name { get; }
+
+        public ImmutableArray<DocumentSymbolItem> Children { get; set; }
+
+        public Position StartPosition { get; }
+        public Position EndPosition { get; }
```

Comment:
as per previous comment, this should become just a normal roslyn TextSpan object.